### PR TITLE
Require password confirmation before setting/removing superuser access.

### LINKED ISSUE #13710: Personal token auth should be only visible after entering the password
Currently, you only need to click to view the token auth. I reckon instead we should ask a user to enter the password again before showing it. We could potentially always show the first 5 characters though. from @tsteur

### LINKED ISSUE #13711: Require password confirmation when giving someone super user access
Similar to #2932 and https://github.com/matomo-org/matomo/issues/13710 it would be also good to eventually have a password confirmation when giving someone user access

### PR DESCRIPTION
**Pull request by diosmosis**
Adds the confirmation to listAllAPI.twig and to userSettings.twig. There is some code redundancy between the two which I couldn't find a good way of removing.

Fixes #13711

### CODE REVIEW

**Comment by tsteur** [comment]
Not sure if it's needed in `listAllApi`? You could also see the token in export container, or just by typing `piwik.token_auth` or something in the console... also by monitoring network requests etc. I reckon we would only do it in the personal settings screen for now? Or maybe we even don't do it at all until we removed it everywhere and replaced token_auth with something better?

**Comment by diosmosis** [comment]
cc @mattab

**Comment by mattab** [comment]
> Or maybe we even don't do it at all until we removed it everywhere and replaced token_auth with something better?

Sounds good to me :+1: 

 better API authentication will be done later in https://github.com/matomo-org/matomo/issues/6559

**Comment by diosmosis** [comment]
Closing

**Comment by diosmosis** [comment]
Whoops forgot there was another change here

**Comment by diosmosis** [comment]
@tsteur removed the code to require password before showing token auth. UI tests may need to be fixed.

File: plugins/UsersManager/API.php [hunk 1]
```diff
@@ -698,12 +698,14 @@ public function addUser($userLogin, $password, $email, $alias = false, $_isPassw
      * @param string   $userLogin          the user login.
      * @param bool|int $hasSuperUserAccess true or '1' to grant Super User access, false or '0' to remove Super User
      *                                     access.
+     * @param string $passwordConfirmation the current user's password.
      * @throws \Exception
      */
-    public function setSuperUserAccess($userLogin, $hasSuperUserAccess)
+    public function setSuperUserAccess($userLogin, $hasSuperUserAccess, $passwordConfirmation)
```
Comment (sgiehl) [inline plugins/UsersManager/API.php]: I guess we should mention that *breaking* API change in the changelog

File: plugins/UsersManager/API.php [hunk 2]
```diff
@@ -698,12 +698,14 @@ public function addUser($userLogin, $password, $email, $alias = false, $_isPassw
...
     {
         Piwik::checkUserHasSuperUserAccess();
         $this->checkUserIsNotAnonymous($userLogin);
+        $this->confirmCurrentUserPassword($passwordConfirmation);
```
Comment (sgiehl) [inline plugins/UsersManager/API.php]: That method actually doesn't exists (anymore?)

**Review by sgiehl — CHANGES_REQUESTED** [review]
Due to the not existing method confirming the password currently returns a `500 - internal server error`. Wondering if we should handle such issues in any way. Currently the popover is closed and nothing happens. Maybe it would be good to at least display an error message?

File: plugins/UsersManager/API.php (hunk 2 above)
Comment (diosmosis) [reply plugins/UsersManager/API.php]: Added it, maybe I forgot to commit...

File: plugins/UsersManager/API.php (hunk 1 above)
Comment (diosmosis) [reply plugins/UsersManager/API.php]: Added to the changelog

**Comment by diosmosis** [comment]
CC @tsteur to make sure the breaking API change is ok

**Comment by tsteur** [comment]
This breaking change should be fine 👍

**Comment by tsteur** [comment]
Looks good. 
I looked at dependencies where we use this method and I think it be good if we could maybe disable the passwordConfirmation check when we are in CLI mode? This would eg help on the cloud.

Aso I think `Installation\Controller::createSuperUser` needs to be adjusted and some tests need to be fixed.

**Pull request merged**